lib/mdl/rules.rb

```diff
@@ -163,9 +163,24 @@
 
 rule "MD013", "Line length" do
   tags :line_length
-  params :line_length => 80
+  params :line_length => 80, :code_blocks => true, :tables => true
   check do |doc|
-    doc.matching_lines(/^.{#{@params[:line_length]}}.*\s/)
+    # Every line in the document that is part of a code block.
+    codeblock_lines = doc.find_type_elements(:codeblock).map{
+      |e| (doc.element_linenumber(e)..
+           doc.element_linenumber(e) + e.value.count('\n') - 1).to_a }.flatten
+    # Every line in the document that is part of a table.
+    locations = doc.elements
+                .map { |e| [e.options[:location], e] }
+                .reject { |l, _| l.nil? }
+    table_lines = locations.map.with_index {
+      |(l, e), i| (i + 1 < locations.size ?
+                   (l..locations[i+1].first - 1) :
+                   (l..doc.lines.count)).to_a if e.type == :table }.flatten
```

Comment:
Given that tables are nesting block elements, finding the line range that makes up a table is done by looking for the next element with a location, or if none exist, deducing it is the last one, hence the end of the file.